### ISSUE TITLE
Prevent undefined object errors (UndefinedObject) not accounting for default value

### ISSUE THREAD

**Issue by qedric** [Shopify/theme-check#527]
<div class="{{ class | default: '' }}">

This is causing a false positive for the undefined object error. If there's a default pipe, surely this should be ignored?

**Comment by charlespwd** [comment]
An oversight. I'm assuming this comes from a snippet file and the `class` param can be omitted?

**Comment by charlespwd** [comment]
In the meantime, you can do one of the following:

- [Disable the rule for snippets with the `exclude_snippets` option](https://github.com/Shopify/theme-check/blob/main/docs/checks/undefined_object.md#exclude_snippets)
- Disable the rule for that file or section with the `theme-check-disable UndefinedObject` comment.

**Comment by qedric** [comment]
Will do, thanks :)

**Comment by charlespwd** [comment]
Still going to fix it, but just learned from a unit test that this wouldn't flag an error:

```liquid
{% assign class = class | default: '' %}
<p class="{{ class }}">...</p>
```

It's one way to declare your variables. I still think the magic comment would be better though. But this has the benefit of already being supported :)
